csrc/quantization/quant_llm/quant_llm_linear.cu

```diff
@@ -0,0 +1,302 @@
+            Split_K);
+        break;
+      case 16:
+        Kernel_Ex<TilingConfig<4, 1, 2>, half, EXPONENT, MANTISSA>(
+            stream, Weight, Scales, B, C, M_Global, N_Global, K_Global,
+            Split_K);
+        break;
+      case 32:
+        Kernel_Ex<TilingConfig<4, 1, 4>, half, EXPONENT, MANTISSA>(
+            stream, Weight, Scales, B, C, M_Global, N_Global, K_Global,
+            Split_K);
+        break;
+      case 64:
+        Kernel_Ex<TilingConfig<4, 1, 8>, half, EXPONENT, MANTISSA>(
+            stream, Weight, Scales, B, C, M_Global, N_Global, K_Global,
+            Split_K);
+        break;
+      case 128:
+        Kernel_Ex<TilingConfig<4, 1, 8>, half, EXPONENT, MANTISSA>(
+            stream, Weight, Scales, B, C, M_Global, N_Global, K_Global,
+            Split_K);
+        break;
+      default:
+        if (N_PowerOf2 % 128 != 0) {
+          printf("FP6LLM_API Error: Unsupported N dimension %d!\n", N_PowerOf2);
+          return cudaErrorUnknown;
+        }
+        Kernel_Ex<TilingConfig<4, 1, 8>, half, EXPONENT, MANTISSA>(
+            stream, Weight, Scales, B, C, M_Global, N_Global, K_Global,
+            Split_K);
+        break;
+    }
+  } else {
+    switch (N_PowerOf2) {
+      case 8:
+        Kernel_Ex<TilingConfig<4, 1, 1>, float, EXPONENT, MANTISSA>(
+            stream, Weight, Scales, B, Reduction_Workspace, M_Global, N_Global,
+            K_Global, Split_K);
+        break;
+      case 16:
+        Kernel_Ex<TilingConfig<4, 1, 2>, float, EXPONENT, MANTISSA>(
+            stream, Weight, Scales, B, Reduction_Workspace, M_Global, N_Global,
+            K_Global, Split_K);
+        break;
+      case 32:
+        Kernel_Ex<TilingConfig<4, 1, 4>, float, EXPONENT, MANTISSA>(
+            stream, Weight, Scales, B, Reduction_Workspace, M_Global, N_Global,
+            K_Global, Split_K);
+        break;
+      case 64:
+        Kernel_Ex<TilingConfig<4, 1, 8>, float, EXPONENT, MANTISSA>(
+            stream, Weight, Scales, B, Reduction_Workspace, M_Global, N_Global,
+            K_Global, Split_K);
+        break;
+      case 128:
+        Kernel_Ex<TilingConfig<4, 1, 8>, float, EXPONENT, MANTISSA>(
+            stream, Weight, Scales, B, Reduction_Workspace, M_Global, N_Global,
+            K_Global, Split_K);
+        break;
+      default:
+        if (N_PowerOf2 % 128 != 0) {
+          printf("FP6LLM_API Error: Unsupported N dimension %d!\n", N_PowerOf2);
+          return cudaErrorUnknown;
+        }
+        Kernel_Ex<TilingConfig<4, 1, 8>, float, EXPONENT, MANTISSA>(
+            stream, Weight, Scales, B, Reduction_Workspace, M_Global, N_Global,
+            K_Global, Split_K);
+        break;
+    }
+    // Reduction for SplitK
+    dim3 GridDim((M_Global * N_Global) / REDUCTION_ELEMENT_PER_THREADBLOCK, 1,
+                 1);
+    dim3 BlockDim(WARP_SIZE, 1, 1);
+    SplitK_Reduction<<<GridDim, BlockDim, 0, stream>>>(
+        C, Reduction_Workspace, M_Global, N_Global, Split_K);
+  }
```

Comment:
Should this be skipped `if (Split_K == 1)`

Reply:
AFAIK there aren't any partial results to combine if `Split_K == 1` (multiplication isn't split), so we don't need reduction for it. The output `C` from `QUANT_GEMM_Kernel` should be the complete result.